File: daily/1431/solution.js
// 题目来源：https://leetcode-cn.com/problems/kids-with-the-greatest-number-of-candies/
var kidsWithCandies = function(candies, extraCandies) {
    var n = candies.length;
    var ans = [];
    var max = 0;
    for(var i = 0; i < n; i++) {
        max = Math.max(max, candies[i]);
    }

    for(var i = 0; i < n; i++) {
        if(candies[i] + extraCandies >= max) ans.push(true);
        else ans.push(false);
    }

    return ans;
};

File: daily/1431/solution.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        if(candies.empty()) return {};

        int maxVal = 0;
        for(int i = 0; i < candies.size(); ++i){
            maxVal = max(candies[i], maxVal);
        }

        vector<bool> ans;
        for(int i = 0; i < candies.size(); ++i){
            if(maxVal-candies[i] <= extraCandies){
                ans.push_back(true);
            }else ans.push_back(false);
        }
        return ans;
    }
};